cmd/natsresolverconfigbuilder.go

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2018-2020 The NATS Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/nats-io/jwt"
+)
+
+type NatsResolverConfigBuilder struct {
+	operator       string
+	operatorName   string
+	sysAccountSubj string
+	sysAccount     string
+	sysAccountName string
+}
+
+func NewNatsResolverConfigBuilder() *NatsResolverConfigBuilder {
+	cb := NatsResolverConfigBuilder{}
+	return &cb
+}
+
+func (cb *NatsResolverConfigBuilder) Add(rawClaim []byte) error {
+	token := string(rawClaim)
+	gc, err := jwt.DecodeGeneric(token)
+	if err != nil {
+		return err
+	}
+	switch gc.Type {
+	case jwt.OperatorClaim:
+		if claim, err := jwt.DecodeOperatorClaims(token); err != nil {
+			return err
+		} else {
+			cb.operator = token
+			cb.operatorName = claim.Name
+		}
+	case jwt.AccountClaim:
+		if claim, err := jwt.DecodeAccountClaims(token); err != nil {
+			return err
+		} else if claim.Subject == cb.sysAccountSubj {
+			cb.sysAccount = token
+			cb.sysAccountName = claim.Name
+		}
+	}
+	return nil
+}
+
+func (cb *NatsResolverConfigBuilder) SetOutputDir(fp string) error {
+	return errors.New("nats-resolver configurations don't support directory output")
+}
+
+func (cb *NatsResolverConfigBuilder) SetSystemAccount(id string) error {
+	cb.sysAccountSubj = id
+	return nil
+}
+
+const tmpl = `# Operator named %s
+operator: %s
+# System Account named %s
+system_account: %s
+
+# configuration of the nats based resolver
+resolver {
+    type: full
+    # Directory in which the account jwt will be stored
+    dir: './jwt'
```

Comment:
Changing to double quotes to be consistent with `interval`:

```suggestion
    dir: "./jwt"
```

Reply:
fixed